localstack-core/localstack/services/events/provider.py

```diff
@@ -1364,22 +1366,96 @@ def _process_entry(
                     }
                 )
             )
-            return
+            return {"EventId": event_formatted["id"]}
+
         self._proxy_capture_input_event(event_formatted)
-        if configured_rules := list(event_bus.rules.values()):
-            for rule in configured_rules:
-                self._process_rules(
-                    rule, region, account_id, event_formatted, processed_entries, failed_entry_count
-                )
-        else:
+
+        has_processing_error = self._process_rules_for_event(
+            event_bus.rules.values(), region, account_id, event_formatted, failed_entry_count
+        )
+
+        if has_processing_error:
+            failed_entry_count["count"] += 1
+            return {
+                "ErrorCode": "InternalException",
+                "ErrorMessage": "Error occurred while processing one or more targets",
+            }
+
+        return {"EventId": event_formatted["id"]}
+
+    def _process_rules_for_event(
+        self,
+        rules: list[Rule],
+        region: str,
+        account_id: str,
+        event_formatted: FormattedEvent,
+        failed_entry_count: dict[str, int],
+    ) -> bool:
+        """Process all rules for an event. Returns True if any processing error occurred."""
+        has_error = False
+
+        if not rules:
             LOG.info(
                 json.dumps(
                     {
                         "InfoCode": "InternalInfoEvents at process_rules",
-                        "InfoMessage": f"No rules attached to event_bus: {event_bus_name}",
+                        "InfoMessage": "No rules configured for event bus",
                     }
                 )
             )
+            return has_error
+
+        for rule in rules:
+            event_pattern = rule.event_pattern
```

Comment:
we need to align these changes with EventStudio before we merge this - since we are patching specific methods and these changes here will break things.